### PR TITLE
Dispatch source

### PR DESCRIPTION
**Pull request by jkanem**
Fixes https://github.com/AgnostiqHQ/covalent/issues/80 as well as gathers imports in the dispatch_source.py file, and comments out Covalent-related decorators.

- [ ] I have added the tests to cover my changes.
- [x] I have updated the documentation, VERSION, and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

### CODE REVIEW

File: covalent/_results_manager/result.py [hunk 1]
```diff
@@ -467,17 +473,54 @@ def _write_dispatch_to_python_file(self, directory: str = None) -> None:
...
+
+        has_cova_decorator = False
+        in_decorator = 0
+        function_lines = function_string.split("\n")
+        for i in range(len(function_lines)):
+            line = function_lines[i].strip()
+            if in_decorator > 0:
+                function_lines[i] = f"# {function_lines[i]}"
+                in_decorator += line.count("(")
+                in_decorator -= line.count(")")
```
Comment (kessler-frost) [inline covalent/_results_manager/result.py:514]: Codecov says this line is not covered by the tests. Can you check that as well?

**Review by kessler-frost — COMMENTED** [review]
Looks good, only some small changes.

File: covalent/_results_manager/result.py [hunk 2]
```diff
@@ -439,11 +439,14 @@ def _write_dispatch_to_python_file(self, directory: str = None) -> None:
...
-            f"# File created by Covalent using covalent version {pkg_resources.get_distribution('cova').version}\n"
-            f"# Note that this file does not contain any global imports you did in your original dispatch\n"
-            f"# Covalent result -"
-        )
+        dispatch_function = f"# File created by Covalent using covalent version {pkg_resources.get_distribution('cova').version}\n"
+
+        dispatch_function += f"# Result of dispatch {self.dispatch_id}\n"
+        dispatch_function += f"# Result status: {self.status}\n"
+        dispatch_function += f"# Result start time: {self.start_time}\n"
+        dispatch_function += f"# Result end time: {self.end_time}" + "\n"
```
[the remark this answers is not in the document]
Comment (jkanem) [reply covalent/_results_manager/result.py]: Done

**Review by santoshkumarradha — CHANGES_REQUESTED** [review]
@kessler-frost can you add your graph to the main issue just fine we have a record of it.

**Review by santoshkumarradha — APPROVED** [review]

File: covalent/_results_manager/result.py (hunk 1 above)
Comment (jkanem) [reply covalent/_results_manager/result.py:526]: I've added a test for the entire function.

**Review by kessler-frost — APPROVED** [review]

**Pull request merged**